theano/sandbox/cuda/basic_ops.py

```diff
@@ -3041,24 +3041,6 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
-        # Get joining axis as int
```

Comment:
Keep that in the make_join_internal. This help canolicalize the ops and help the merge when there is constant.